src/config/field-overrides/program-indicator/OrgUnitField.js

```diff
@@ -0,0 +1,323 @@
+- This field component is dependant on the values in the `program`
+  and `analyticsType` field
+- It needs to be aware of when these values change and update its UI
+- Ideally, it would have been good if we could have specified the following
+  two field rules:
+  1. When program changes set the `programId` prop of the `OrgUnitField`
+  2. When analyticsType changes set the `analyticsType` prop of the `OrgUnitField`
+- But as far as I could tell, our field rules do not support this and I did not
+  see an easy way to add support either. So trying to implement this in the
+  `componentDidUpdate` hook of the component seemed like the only way forward.
+- Without a field-rule in place for the `orgUnitField`, the `componentDidUpdate`
+  hook will not get called when other fields change. So we need a field rule and
+  luckily, the issue also specifies that the `orgUnitField` needs to be hidden
+  when no program is selected yet.
+- So `componentDidUpdate` is being called because the field-rule is in place.
+  We must not remove it.
+- And unfortunately, we are reading the `programId` and `analyticsType` from a
+  d2 model. Since these are mutable objects the values we find in `prevProps.model`
+  are identical to the ones in `this.props.model`. To address this issue we have to
+  keep track of the previous values manually
+*/
+
+export default class OrgUnitField extends Component {
+    constructor(props, context) {
+        super(props);
+        this.d2 = context.d2;
+        this.translate = this.d2.i18n.getTranslation.bind(context.d2.i18n);
+        this.state = {
+            loading: false,
+            error: null,
+            value: undefined,
+            dataElements: {},
+        };
+        this.staticOptions = {
+            event: {
+                value: ANALYTICS_TYPE_EVENT,
+                text: this.translate('Event'),
+            },
+            enrollment: {
+                value: ANALYTICS_TYPE_ENROLLMENT,
+                text: this.translate('Enrollment'),
+            },
+            ownerAtStart: {
+                value: 'OWNER_AT_START',
+                text: this.translate('Owner at start'),
+            },
+            ownerAtEnd: {
+                value: 'OWNER_AT_END',
+                text: this.translate('Owner at end'),
+            },
+            registration: {
+                value: 'REGISTRATION',
+                text: this.translate('Registration'),
+            },
+        };
+        this.onChange = this.onChange.bind(this);
+
+        this.updateManualPrevProps(props);
+    }
+
+    componentDidMount() {
+        if (this.hasRequiredParams()) {
+            this.fetchDataElements();
+        }
+
+        this.updateManualPrevProps();
+    }
+
+    componentDidUpdate() {
+        if (
+            this.hasRelevantPropChanged() &&
+            this.shouldFetchDataElementsForProgram()
+        ) {
+            this.fetchDataElements();
+        }
+
```

Comment:
Currently there's a bug causing the old `value` to linger if you change the program. And in some cases this value shouldn't be a valid selection if program is changed.

So I think something like this is needed:
```js
      if(this.hasProgramChanged()) {
            this.onChange({target: { value: this.getDefaultValue()}})
        }
```

Reply:
After discussing some details and edge cases with Birk on Slack. I've implemented the following changes:
* I am holding on to the selected value, unless it's not among the new dropdown-options, in which case I reset to default (so basically option 2 above).
* Since all of this resetting can be a bit confusing to a user who may or may not be aware of what is going on behind the scenes, I thought it would be beneficial to add a `(default)` suffix in the labels of the default options, so that is clear for users that we're resetting to the default. 
* This also means that we now have two "flavours" of enrolment options: 
    * `Enrollment organisation unit (default)` which is the default when `analyticsType === ENROLLMENT`
    * `Enrollment organisation unit` which is an option the user may select explicitly when `analyticsType === EVENT`